### PR TITLE
TRAFODION [1879] - Integrate Library Management into Trafodion Metadata

### PR DESCRIPTION
**Pull request by robertamarton**
The initial release of library management has been delivered to Trafodion. This
is follow-on work that integrates library management operations into the
existing Trafodion infrastructure. Currently, with the initial release of
library management, the consumer needs to run a special script to set up
everything. This delivery incorporates the steps into existing SQL commands and
removes the need for the script.

### CODE REVIEW

**Comment by robertamarton** [comment]
jenkins, retest

File: core/sql/common/ComUser.cpp [hunk 1]
```diff
@@ -347,3 +347,57 @@ Int16 ComUser::getAuthNameFromAuthID(Int32   authID,
...
+//   (out) actualLen - the length of the returned role list
+//   ( in) maxLen - the size of the roleList allocated by the caller
+//   ( in) delimited - delimiter to use (defaults to single quote)
+//   ( in) separator - specified what separator to use (defaults to comma)
+//   ( in) includeSpecialAuths - includes the special auths (PUBLIC and _SYSTEM) 
+//
+//  Returns:  FEOK -- found
+//            FEBUFTOOSMALL -- space allocated for role list is too small 
+// ----------------------------------------------------------------------------
+Int32 ComUser::getRoleList (char * roleList,
```
Comment (DaveBirdsall) [inline core/sql/common/ComUser.cpp]: I wonder if we could return an NAString instead, then we'd never have to trouble ourselves with FEBUFTOOSMALL? (At the price of dynamic heap management though.)

File: core/sql/sqlcomp/CmpSeabaseDDLroutine.cpp [hunk 2]
```diff
@@ -1425,3 +1426,169 @@ short CmpSeabaseDDL::validateRoutine(ExeCliInterface *cliInterface,
...
+                         gluedQuery, gluedQuerySize);
+
+      param_[0] = getSystemCatalog();
+      param_[1] = SEABASE_LIBMGR_SCHEMA;
+      param_[2] = getSystemCatalog();
+      param_[3] = SEABASE_LIBMGR_SCHEMA;
+      param_[4] = SEABASE_LIBMGR_LIBRARY;
+
+      str_sprintf(queryBuf, gluedQuery, param_[0], param_[1], param_[2], param_[3], param_[4]);
+      NADELETEBASIC(gluedQuery, STMTHEAP);
```
Comment (DaveBirdsall) [inline core/sql/sqlcomp/CmpSeabaseDDLroutine.cpp]: How do we know that queryBuf is large enough to hold the data str_sprintf writes to it?

**Comment by DaveBirdsall** [comment]
Looks good to me.

File: core/sql/common/ComUser.cpp (hunk 1 above)
Comment (robertamarton) [inline core/sql/common/ComUser.cpp]: I thought about it, I tried to keep the code similar to other functions in the same file.  But you have a point. Will think about it for a subsequent release.

File: core/sql/sqlcomp/CmpSeabaseDDLroutine.cpp (hunk 2 above)
Comment (robertamarton) [inline core/sql/sqlcomp/CmpSeabaseDDLroutine.cpp]: 2000 is big enough but I changed the approximate size calculation based on the size of the catalog, schema, library, and glued query - will deliver in my next release - coming soon.

**Pull request merged**